FEM/fem_ele_vec.cpp

```diff
@@ -1325,6 +1298,9 @@ void CFiniteElementVec::GlobalAssembly_Stiffness()
 	if (act_nodes_h != nnodesHQ)
 		non_ghost_flag = -1;
 
+	petsc_group::PETScLinearSolver* eqs = pcs->eqs_new;
+
+	const PetscInt zero_id_alias = -eqs->Size();
```

Comment:
The negative indices are used here for ghost entries. Since zero can not be assigned as negative and an entry with ID of zero could be ghost, the number of -eqs->Size() is selected as an alias of ID of zero for ghost entry. Where,  -eqs->Size()  is just the largest entry ID plus 1.

Actually, the bug existed in this function before.